frontend/cypress/e2e/exercise.cy.ts

```diff
@@ -0,0 +1,851 @@
+
+        cy.wait(1000);
+        cy.get('@trainerSocketPerformedActions')
+            .lastElement()
+            .should('have.property', 'type', '[Exercise] Tick');
+
+        cy.getState()
+            .its('exerciseState')
+            .its('currentTime')
+            .should('be.greaterThan', 0);
+
+        cy.log('pause an exercise')
+            .get('[data-cy=trainerToolbarPauseButton]')
+            .click();
+
+        cy.get('@trainerSocketPerformedActions')
+            .atPosition(-2)
+            .should('have.property', 'type', '[Exercise] Pause');
+
+        cy.get('@trainerSocketPerformedActions')
+            .lastElement()
+            .should(
+                'have.property',
+                'type',
+                '[Emergency Operation Center] Add Log Entry'
+            );
+
+        cy.getState()
+            .its('exerciseState')
+            .should('have.property', 'currentStatus', 'paused');
+    });
+
+    it('can manage participants', () => {
+        cy.initializeParticipantSocket();
+
+        cy.get('@trainerSocketPerformedActions')
+            .lastElement()
+            .should('have.property', 'type', '[Client] Add client');
+
+        cy.getState()
+            .its('exerciseState')
+            .its('clients')
+            .itsValues()
+            .should('have.length', 3);
+
+        cy.dragToMap('[data-cy=draggableViewportDiv]');
+
+        cy.get('[data-cy=trainerToolbarExecutionButton]').click();
+        cy.get('[data-cy=trainerToolbarParticipantsButton]').click();
+
+        cy.get('[data-cy=clientPopupSetWaitingRoomCheckbox]').last().check();
+
+        cy.get('@participantSocketPerformedActions')
+            .lastElement()
+            .should('have.property', 'type', '[Client] Set waitingroom');
+
+        cy.get('@participantSocketUUID', { log: false }).then(
+            (participantSocketUUID: any) => {
+                cy.getState()
+                    .its('exerciseState')
+                    .its('clients')
+                    .atKey(participantSocketUUID)
+                    .should('have.property', 'isInWaitingRoom', false);
+            }
+        );
+
+        cy.get('[data-cy=clientPopupSetViewportButton]').last().click();
+        cy.get('[data-cy=clientPopupSetViewportDropdownButton]:visible')
+            .first()
+            .click();
+
+        cy.get('@participantSocketPerformedActions')
+            .lastElement()
+            .should('have.property', 'type', '[Client] Restrict to viewport');
+
+        cy.get('@participantSocketUUID', { log: false }).then(
+            (participantSocketUUID: any) => {
+                cy.getState()
+                    .its('exerciseState')
+                    .its('clients')
+                    .atKey(participantSocketUUID)
+                    .should('have.property', 'viewRestrictedToViewportId');
+            }
+        );
+
+        cy.get('@participantSocket', { log: false }).invoke('disconnect');
+
+        cy.wait(0);
```

Comment:
What does this do?

Reply:
It waits for the socket to disconnect. It's there for a reason, but I don't know if I completely understand why.